apps/gateway/src/middlewares/ai-models/providers/vertex.ts

```diff
@@ -0,0 +1,95 @@
+import { createVertex } from "@ai-sdk/google-vertex";
+
+import type { GoogleProviderConfig } from "@hebo/database/src/types/providers";
+import { getSecret } from "@hebo/shared-api/utils/secrets";
+
+import type { AwsTemporaryCredentials } from "./bedrock";
+import type { Provider } from "./types";
+import type { Provider as AiProvider } from "ai";
+
+type AwsContainerCredentialResponse = {
+  AccessKeyId: string;
+  SecretAccessKey: string;
+  Token: string;
+};
+
+const toTemporaryCredentials = (
+  credentials: AwsContainerCredentialResponse,
+): AwsTemporaryCredentials => ({
+  accessKeyId: credentials.AccessKeyId,
+  secretAccessKey: credentials.SecretAccessKey,
+  sessionToken: credentials.Token,
+});
+
+// FUTURE: Let google auth library handle this once they will start supporting WIF for ECS tasks: https://github.com/googleapis/google-auth-library-php/issues/496
+const injectAwsMetadataCredentials = async () => {
```

Comment:
Code that is specific to vertex should be in the VertexProvider class (instead of standalone functions). Code that is specific to the gateway itself running on AWS should be in a separate aws adapter module. In the future we might want to implement other adapters like vercel, cloudflare, ...